models.go

```diff
@@ -11,4 +13,16 @@ type (
 type Document struct{}
 
 // Collection represents an outline collection.
-type Collection struct{}
+type Collection struct {
```

Comment:
Do we really want to return this struct to the callers? I ask because the struct contains json information, therefore it looks more like an "Dto" or "Api" model than an domain model. Bus as we return it it should rather be an domain model...http information should be abstracted away IMHO... No? 🤔

Reply:
Would you care to elaborate what such a domain model may look like?

Reply:
I thought about this again and came to the conclusion that this is fine as it is 🙃

We are building mainly an API client therfore it is kind og obvious that objects may contain json information...
And in case such json objects breaks because the api changed, we could simply do an major release of the project 🙃.

So ignore me 🙃. Everything is fine 👌

Reply:
Haha, exactly my thinking.

From my understanding our objective is to automate manual steps for communicating with outline's HTTP API to take care of daily chores which we do manually with a tool like Postman:
- Serializing and de-serializing objects correctly
- Making sure correct URL is called for specific resources
- Correct headers, tokens
- etc.

So our client essentially provides a thin/passthrough API where 'view' models are the 'domain' models. Whatever business logic we have operates on these only. Any breaking changes in upstream API spec would need to be passed to the user transparently hence separate domain models would be an unnecessary indirection. 

The client/cli packages have almost 1:1 relation to the API we are trying to abstract. We are not adding extra 'business value' on top of this simple transactions.

In contrast lets say we were building AWS client/cli for starting a VM and a Lambda function. Lets say we decide to provide extra experience on top of regular API transactions: **a progress bar**. So polling a VM startup or lambda function would be different. We would poll VM creation much less frequently than lambda function because the two have very different cycles. So we would need internal metadata (aka some fancy domain models) which would update 'progress view' for a VM creation differently than that for a lambda function creation.

Coming back to our client right now our aim is to improve and stabilize this thin wrapper at least for v0.1.0 and may be for a few more iterations. After then if we decide that we can add more 'value' on top of this wrapper then of course we would make changes. And since everything would  still be 0.x, making breaking changes is no problem.

Reply:
Nice explanation! Nice example. Thank you very much. Really appreciate!